src/Server/TextDocument.php

```diff
@@ -265,7 +265,15 @@ public function references(
     public function definition(TextDocumentIdentifier $textDocument, Position $position): Promise
     {
         return coroutine(function () use ($textDocument, $position) {
-            $document = yield $this->documentLoader->getOrLoad($textDocument->uri);
+            $documentLoader = $this->documentLoader;//->getOrLoad($textDocument->uri);
+            $document = null;
+            if (isset($documentLoader->documents[$textDocument->uri])) {
```

Comment:
Gotcha. A client is expected to call `textDocument/didOpen` if we wants to do many operations on a specific file, and close it with `textDocument/didClose` once he's done with it.